[Regression] Missing type declarations in C code if members are used in {.emit.}

Introduced in 70b28a39fedfe030ac4e1615d32328d7ffd3f79d.

```nim
type Foo {.exportc.} = object
  a: cint
proc f(foo: ptr Foo): cint =
  {.emit: "`result` = `foo`->a;".}
echo f(nil)
```
```
Error: execution of an external compiler program 'clang -c  -w  -I/Users/yglukhov/Projects/nim/lib -o /Users/yglukhov/nimcache/test.o /Users/yglukhov/nimcache/test.c' failed with exit code: 1

/Users/yglukhov/nimcache/test.c:82:14: error: incomplete definition of type 'struct Foo'
        result = foo->a;
                 ~~~^
/Users/yglukhov/nimcache/test.c:24:16: note: forward declaration of 'struct Foo'
typedef struct Foo Foo;
               ^
1 error generated.
```
@cooldome, can we fix it so that `{.exportc.}` types are always declared completely?

--- a/doc/manual/stmts.txt
+++ b/doc/manual/stmts.txt
@@ -145,7 +145,7 @@ the variable has been initialized and does not rely on syntactic properties:
       x = a()
     else:
       x = a()
-    use x
+    # use x
 
 
 let statement
@@ -296,7 +296,7 @@ empty ``discard`` statement should be used.
 For non ordinal types it is not possible to list every possible value and so
 these always require an ``else`` part.
 
-As case statements perform compile-time exhaustiveness checks, the value in 
+As case statements perform compile-time exhaustiveness checks, the value in
 every ``of`` branch must be known at compile time. This fact is also exploited
 to generate more performant code.
 

